File: 30-Days_Data_Structures/Day-04_Recursion/Q01-Fibonacci.js
// In fibonaci
// f(n) = f(n-1) + f(n-2)

const fibonacci = (num) => {
  if (num === 0) {
    return 0;
  }
  if (num === 1) {
    return 1;
  }

  return fibonacci(num - 1) + fibonacci(num - 2);
};

// 👆The above method has 2^n time complexity.
// Learn in detail about the time complexity how it is calculated.

File: 30-Days_Data_Structures/Day-04_Recursion/Q01-Fibonacci.py
# fibonacci(n) =  fibonacci(n - 1) + fibonacci(n - 2)

def find_fibonacci(num):
    if num == 0:
        return 0
    if num == 1:
        return 1

    fibonacci = find_fibonacci(num - 1) + find_fibonacci(num - 2)
    return fibonacci


print(find_fibonacci(40))
